relevanceai/operations/dr/base.py

```diff
@@ -82,20 +90,107 @@ def fit_transform_documents(
             A list of documents with the original vector field and the new vector field.
 
         """
-        documents = list(self.filter_docs_for_fields([vector_field], documents))
+
+        documents = list(self.filter_docs_for_fields(vector_fields, documents))
         vectors = np.array(
-            self.get_field_across_documents(
-                vector_field, documents, missing_treatment="skip"
+            self.get_fields_across_documents(
+                vector_fields, documents, missing_treatment="skip"
             )
         )
+        vectors = vectors.reshape(-1, vectors.shape[1] * vectors.shape[2])  # hacky fix
```

Comment:
Flattening again? - can you explain what "hacky fix" means here?

Reply:
Flattening only happens once in this call. fit_documents seems to only be fitting not transforming so i didnt use that. hacky because this wont work for vectors of different length atm.